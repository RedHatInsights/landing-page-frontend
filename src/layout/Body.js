import React, { useState } from 'react';
import {
    PageSection,
    Tabs, Tab
} from '@patternfly/react-core';

import Hero from './Hero';
import BannerCard from '../components/BannerCard';
import FancyLink from '../components/FancyLink';

import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import './Body.scss';

export function Body({ technologies }) {

    const [ activeTabKey, setActiveTabKey ] = useState(0);

    const handleTabClick = (event, tabIndex) => {
        setActiveTabKey(tabIndex);
    };

    return (
        <React.Fragment>
            <Hero
                title='Scale the management and operations of your IT infrastructure'
                subtitle='Discover purpose-built services exclusively for Red Hat subscribers'
                className='ins-c-hero__small'
                data-ouia-component-type='hero'/>
            <PageSection className='pf-m-fill ins-p-landing__content'>
                <Tabs data-ouia-navigation='true' activeKey={ activeTabKey } isSecondary onSelect={ handleTabClick }>
                    { technologies.map(({ title, id, description, image, apps }, key) => ( // map categories
                        <Tab key={ key }
                            data-ouia-component-type='tab'
                            data-ouia-component-id={ `nav-tab-${id}` }
                            data-ouia-navigation-name={ `Tab ${id}` }
                            eventKey={ key || 0 }
                            title={ <BannerCard title={ title } category-id={ id } description={ description } image={ image }/> }>
                            <div className='ins-l-app-grid'>
                                { apps.map(({ name, url, id, disabled }, key) => (<React.Fragment key={ key }>
                                    { !disabled && <FancyLink
                                        className='ins-l-app-grid__item'
                                        data-ouia-component-type='app-link'
                                        data-ouia-component-id={ `app-link-${id}` }
                                        application-id={ id }
                                        to={ `${window.location.origin}${window.insights.chrome.isBeta() ? '/beta/' : '/'}${url}` }
                                        key={ key }>
                                        { name }
                                    </FancyLink> }
                                </React.Fragment>
                                )) }
                            </div>
                        </Tab>
                    )) }
                </Tabs>
            </PageSection>
        </React.Fragment>
    );
}

Body.propTypes = {
    technologies: PropTypes.arrayOf(PropTypes.shape({
        title: PropTypes.string,
        description: PropTypes.string,
        apps: PropTypes.arrayOf(PropTypes.shape({
            id: PropTypes.string,
            name: PropTypes.string,
            url: PropTypes.string
        }))
    }))
};

Body.defaultProps = {
    technologies: []
};

function mapStateToProps({ technologies } = { technologies: { activeTechnologies: []}}) {
    return {
        technologies: technologies && technologies.activeTechnologies.filter(({ disabled }) => !disabled)
    };
}

export { mapStateToProps };
export default connect(mapStateToProps)(Body);
