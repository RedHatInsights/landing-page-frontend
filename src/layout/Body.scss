// Body.scss
.ins-p-landing__content[landing-page-type="authenticated"] {
    width: 100%;
    margin-left: auto;
    margin-right: auto;

    @media screen and (min-width: 1200px) {
        width: 80%;
    }

    .pf-c-button.pf-m-link.pf-m-inline {
        text-align: left;
    }

    [application-id="insights"] .ins-c-application-info__content-applist {
        display: grid;
        grid-template-columns: auto;
        grid-template-columns: auto auto;
        gap: 10px;
        a { margin-top: 0; }
    }
}

