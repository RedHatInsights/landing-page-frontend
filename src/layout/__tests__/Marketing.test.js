import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Body from '../Marketing';

jest.mock('../../consts', () => {
  const insightsMarketing = require('../../components/marketing/insightsMarketing.svg')
    .default;
  const Insights = require('../../components/Insights.svg').default;

  return {
    activeTechnologies: [
      {
        entitlement: 'test',
        marketing: true,
        marketingImage: insightsMarketing,
        marketingText: 'Marketing Test',
        marketingUrls: {
          learnMore: '/marketing-test',
          tryit: '/tryit-test',
        },
        name: 'test',
        id: 'test',
        url: 'test',
        baseApp: '/test',
        apps: {
          rules: '/app-test',
        },
        icon: Insights,
        title: 'Jest Test',
        emptyTitle: 'test',
        emptyText: 'test',
        emptyAction: {
          title: 'test',
          navigate: '/empty-action-test',
        },
        body: 'test',
      },
    ],
  };
});

describe('render Marketing component', () => {
  it('should render correctly', () => {
    const wrapper = mount(<Body />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
