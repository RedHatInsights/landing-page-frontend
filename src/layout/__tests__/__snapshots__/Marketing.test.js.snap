// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Marketing component should render correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Marketing)>
    <Marketing
      dispatch={[Function]}
      technologies={Array []}
    >
      <Hero>
        <PageSection
          className="ins-c-hero pf-m-fill pf-l-flex pf-m-align-items-center pf-m-justify-content-center"
        >
          <section
            className="pf-c-page__main-section ins-c-hero pf-m-fill pf-l-flex pf-m-align-items-center pf-m-justify-content-center"
          >
            <article
              className="pf-l-flex pf-m-column pf-m-align-items-center"
            >
              <Title
                className="ins-c-hero__title pf-m-spacer-lg"
                headingLevel="h1"
                size="4xl"
              >
                <h1
                  className="pf-c-title pf-m-4xl ins-c-hero__title pf-m-spacer-lg"
                >
                   Explore our open, multicloud tools 
                </h1>
              </Title>
              <Title
                className="ins-c-hero__sub-title pf-m-spacer-xl"
                headingLevel="h2"
                size="xl"
              >
                <h2
                  className="pf-c-title pf-m-xl ins-c-hero__sub-title pf-m-spacer-xl"
                >
                  Discover Red Hat
                  <sup
                    className="ins-c-rball"
                  >
                    Â®
                  </sup>
                   software-as-a-service
                </h2>
              </Title>
              <Component
                className="ins-c-hero__login"
                onClick={[Function]}
              >
                <ComponentWithOuia
                  component={[Function]}
                  componentProps={
                    Object {
                      "children": " Log in to your Red Hat account",
                      "className": "ins-c-hero__login",
                      "onClick": [Function],
                    }
                  }
                  consumerContext={null}
                >
                  <Button
                    className="ins-c-hero__login"
                    onClick={[Function]}
                    ouiaContext={
                      Object {
                        "isOuia": false,
                        "ouiaId": null,
                      }
                    }
                  >
                    <button
                      aria-disabled={null}
                      aria-label={null}
                      className="pf-c-button pf-m-primary ins-c-hero__login"
                      disabled={false}
                      onClick={[Function]}
                      tabIndex={null}
                      type="button"
                    >
                       Log in to your Red Hat account
                    </button>
                  </Button>
                </ComponentWithOuia>
              </Component>
              <Component
                className="ins-c-hero__new-customer"
                component="a"
                href="https://www.redhat.com/en/customers"
                isInline={true}
                variant="link"
              >
                <ComponentWithOuia
                  component={[Function]}
                  componentProps={
                    Object {
                      "children": "Not a customer?",
                      "className": "ins-c-hero__new-customer",
                      "component": "a",
                      "href": "https://www.redhat.com/en/customers",
                      "isInline": true,
                      "variant": "link",
                    }
                  }
                  consumerContext={null}
                >
                  <Button
                    className="ins-c-hero__new-customer"
                    component="a"
                    href="https://www.redhat.com/en/customers"
                    isInline={true}
                    ouiaContext={
                      Object {
                        "isOuia": false,
                        "ouiaId": null,
                      }
                    }
                    variant="link"
                  >
                    <a
                      aria-disabled={false}
                      aria-label={null}
                      className="pf-c-button pf-m-link pf-m-inline ins-c-hero__new-customer"
                      disabled={null}
                      href="https://www.redhat.com/en/customers"
                      tabIndex={null}
                      type={null}
                    >
                      <span
                        className="pf-c-button__text"
                      >
                        Not a customer?
                      </span>
                    </a>
                  </Button>
                </ComponentWithOuia>
              </Component>
            </article>
          </section>
        </PageSection>
      </Hero>
      <PageSection
        className="ins-c-marketing pf-m-no-fill"
        landing-page-type="unauthenticated"
      >
        <section
          className="pf-c-page__main-section ins-c-marketing pf-m-no-fill"
          landing-page-type="unauthenticated"
        >
          <Grid
            gutter="sm"
            md={6}
            sm={12}
            xl={6}
            xl2={4}
          >
            <div
              className="pf-l-grid pf-m-all-12-col-on-sm pf-m-all-6-col-on-md pf-m-all-6-col-on-xl pf-m-all-4-col-on-2xl pf-m-gutter"
            />
          </Grid>
        </section>
      </PageSection>
    </Marketing>
  </Connect(Marketing)>
</Provider>
`;
